src/tools.js

```diff
@@ -42,15 +40,35 @@ const HOP_BY_HOP_HEADERS_REGEX = new RegExp(`^(${HOP_BY_HOP_HEADERS.join('|')})$
 
 export const isHopByHopHeader = (header) => HOP_BY_HOP_HEADERS_REGEX.test(header);
 
+const TOKEN_REGEX = /^[\^_`a-zA-Z\-0-9!#$%&'*+.|~]+$/;
+
+/**
+ * Verifies that the given val is a valid HTTP token per the rules defined in RFC 7230
+ * @see https://tools.ietf.org/html/rfc7230#section-3.2.6
+ * @see https://github.com/nodejs/node/blob/8cf5ae07e9e80747c19e0fc04fad48423707f62c/lib/_http_common.js#L222
+ */
+export const isHttpToken = (val) => TOKEN_REGEX.test(val);
```

Comment:
maybe its better not to export those?

Reply:
Yeah, I'd skip the exports, no need for it..